perfkitbenchmarker/benchmark_spec.py

```diff
@@ -139,59 +195,34 @@ def ConstructVirtualMachines(self):
     """Constructs the BenchmarkSpec's VirtualMachine objects."""
     vm_group_specs = self.config.vm_groups
 
-    zone_index = 0
+    if FLAGS.os_type == os_types.JUJU:
```

Comment:
I noticed a few places where you check FLAGS.os_type - in order to support configs you should check vm.OS_TYPE (or the group spec's os_type) instead (since in some instances FLAGS.os_type may not be os_types.JUJU, but the VM will still have that type).